src/main/java/org/isf/admission/rest/AdmissionController.java

```diff
@@ -212,73 +211,68 @@ public ResponseEntity<List<AdmittedPatientDTO>> allAdmittedPatients()
+	 * Get all the {@link Admission}s that start in the specified range
 	 * 
-	 * @param patientCode
+	 * @param admissionRange
 	 * @return the {@link List} of found {@link Admission} or NO_CONTENT otherwise.
 	 * @throws OHServiceException
 	 */
 	@GetMapping(value = "/admissions", produces = MediaType.APPLICATION_JSON_VALUE)
 	public ResponseEntity<List<AdmissionDTO>> getAdmissions(
-			@RequestParam(name = "patientcode", defaultValue = "0", required = false) int patientcode,
-			@RequestParam(name = "admissionrange", required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime[] admissionRange,
-			@RequestParam(name = "dischargerange", required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime[] dischargeRange)
-			throws OHServiceException {
-		LOGGER.info("Get admissions of patients by id: {}", patientcode);
-		if (admissionRange != null && admissionRange.length == 2) {
-			LOGGER.debug("Get admissions started between {} and {}", admissionRange[0], admissionRange[1]);
-		}
-		if (dischargeRange != null && dischargeRange.length == 2) {
-			LOGGER.debug("Get admissions ended between {} and {}", dischargeRange[0], dischargeRange[1]);
-		}
-		
-		List<AdmittedPatient> admittedPatients = new ArrayList<AdmittedPatient>();
-		if (patientcode == 0) {
-			admittedPatients = admissionManager.getAdmittedPatients(admissionRange, dischargeRange, "");
-		} else {
-			admittedPatients = admissionManager.getAdmittedPatients(admissionRange, dischargeRange, Integer.toString(patientcode));
-		}
-		if (admittedPatients.isEmpty()) {
+					@RequestParam(name = "admissionrange") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime[] admissionRange)
+					throws OHServiceException {
+		LOGGER.debug("Get admissions started between {} and {}", admissionRange[0], admissionRange[1]);
+
+		List<Admission> admissions = admissionManager.getAdmissions(admissionRange[0], admissionRange[1], 0, 0);
+
+		if (admissions.isEmpty()) {
 			return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null);
 		}
-		if (admittedPatients.get(0).getAdmission() == null) {
+		return ResponseEntity.ok(admissionMapper.map2DTOList(admissions));
+	}
+
+	/**
+	 * Get all the {@link Admission}s that end in the specified range
+	 * 
+	 * @param dischargeRange
+	 * @return the {@link List} of found {@link Admission} or NO_CONTENT otherwise.
+	 * @throws OHServiceException
+	 */
+	@GetMapping(value = "/discharges", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<List<AdmissionDTO>> getDischarges(
+					@RequestParam(name = "dischargerange") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime[] dischargeRange)
+					throws OHServiceException {
+		LOGGER.debug("Get admissions that end between {} and {}", dischargeRange[0], dischargeRange[1]);
+
+		List<Admission> admissions = admissionManager.getDischarges(dischargeRange[0], dischargeRange[1], 0, 0);
```

Comment:
Same question here !

Reply:
You're right, can we use these? Is it ok for you?

```
@RequestParam(value="page", required=false, defaultValue="0") Integer page,
@RequestParam(value="size", required=false, defaultValue=DEFAULT_PAGE_SIZE) Integer size) 
```